clustering/__init__.py

```diff
@@ -4,53 +4,25 @@
 from sklearn.decomposition import PCA
 from sklearn.cluster import KMeans
 
-
-def find_incident(labels):
-    res = set()
-    for i in range(len(labels)):
-        if labels[i].startswith('i_'):
-            res.add(i)
-    return res
-
-
-# numpy array construction
-def construct_np_allin(allin, m, n, incident_labels):
-    np_allin = []
-    for i in range(m):
-        tmp = []
-        count = 0
-        for j in range(n):
-            if allin[i][j] == True:
-                tmp.append(1)
-            elif allin[i][j] == False:
-                tmp.append(0)
-            elif allin[i][j] != 'None':
-                tmp.append(1)
-            else:
-                tmp.append(0)
-            if j in incident_labels and allin[i][j] == 1:
-                count += 1
-        np_allin.append(tmp+[count])
-    return np_allin
-
+def _create_incidents_matrix(dataframe):
+    false_values = [False, 'False', None, 'None', 0]
```

Comment:
so  ```np.array([False, 'False', None, 'None', 0]).dtype``` is object, we should be safe here

maybe we want to set it explicitly though, to avoid the bug you see, where ```np.array([False, 'False']).dtype``` is >U4
so: ```np.array([False, 'False', None, 'None', 0], dtype=object)```
(also looking at how to set the exact type, since using object is supposed to be super slow)

Reply:
No need for that... Since there's no longer the need for the numpy conversions I'm gonna use Pandas directly. And that doesn't suffers from the issue, so it's fine.